### ISSUE TITLE
10$: bad expression in 'if' condition causes E0065: 'else' has no corresponding 'if'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1008]
```javascript
    if (e isntanceof DataNotLoadedError) {
    } else {  // E0065: 'else' has no corresponding 'if'
      throw e;
    }
```
quick-lint-js reports E0065 but it should not.
